src/main/java/com/fmi/cardealership/service/UserService.java

```diff
@@ -0,0 +1,82 @@
+package com.fmi.cardealership.service;
+
+import com.fmi.cardealership.exception.EntityNotFoundException;
+import com.fmi.cardealership.model.User;
+import com.fmi.cardealership.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserService {
+    UserRepository userRepository;
+    
+    @Autowired
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    public User addUser(User user) {
+        boolean existsEmail = userRepository
+                .findByEmail(user.getEmail())
+                .isPresent(),
+        existsUsername = userRepository
+                .findByUsername(user.getUsername())
+                .isPresent();
```

Comment:
You can add a unique constraint to those columns. Then you won`t need to check if they are present.